File: src/main/java/SpartJavaSample/ContainerComposer.java
package SpartJavaSample;

import org.picocontainer.DefaultPicoContainer;
import org.picocontainer.MutablePicoContainer;

@SuppressWarnings("unused")
class ContainerComposer
{
    public static void composeApplication(DefaultPicoContainer appContainer)
    {
        appContainer.addComponent(AuthorizeController.class);
        appContainer.addComponent(TokenController.class);
        appContainer.addComponent(LoginController.class);
    }

    public static void composeSession(MutablePicoContainer container) { }

    public static void composeRequest(MutablePicoContainer container) { }
}

File: src/main/kotlin/SpartJavaSample/ContainerComposer.kt
package SpartJavaSample

import org.picocontainer.DefaultPicoContainer
import org.picocontainer.MutablePicoContainer

class ContainerComposer : Composable
{
    public override fun composeApplication(appContainer: MutablePicoContainer)
    {
        appContainer.addComponent(javaClass<AuthorizeController>())
        appContainer.addComponent(javaClass<TokenController>())
        appContainer.addComponent(javaClass<LoginController>())
    }

    public override fun composeRequest(container: MutablePicoContainer)
    {
    }
}
